src/rl/proof_action.py

```diff
@@ -67,6 +67,15 @@ def __ge__(self, other):
             assert isinstance(other, ProofAction.ActionType), f"other must be of type ProofAction.ActionType, not {type(other)}"
             return ProofAction.ActionType.get_order(self) >= ProofAction.ActionType.get_order(other)
 
+    class HammerMode(Enum):
+        NONE = 'NONE' # Prohibit hammer
+        ALLOW = 'ALLOW' # Allow agent to query hammer
+        AUTO = 'AUTO' # Automatically apply hammer (heuristically)
+        ONESHOT = 'ONESHOT' # Proof attempt in one shot using hammer
```

Comment:
Introduce a new Enum, HammerMode, for running different experiments involving hammer.
NONE - hammer throws an error
ALLOW - allow agent to query hammer
AUTO - automatically applies hammer
ONESHOT - only apply hammer to the goal